ydb/core/kqp/provider/yql_kikimr_gateway.h

```diff
@@ -389,6 +390,10 @@ enum EMetaSerializationType : ui64 {
     Json = 2
 };
 
+struct TViewPersistedData {
+    TString QueryText;
```

Comment:
It is SQL. Ideally we would like to save the original text that the user specified at the moment of the CREATE VIEW call. However, I didn't find a way to save the original text (with the original formatting). I save [the query text that is recovered](https://github.com/ydb-platform/ydb/blob/5d59f2cda66c7fe1cc7b09717261af49b87da604/ydb/library/yql/sql/v1/sql_translation.cpp#L4425C52-L4425C52) from the protobuf of the select_stmt message.

Reply:
You should save original query text, since you need to preserve user comments - we support optimiser hints via comments.

You can save original user query as following
1) remember position of first and last SELECT token
2) There is class TTextWalker somewhere in parser code that can help to translate position offsets to byte offset in original query

Also you need to save some parsers settings together with query text. At least ansi_lexer flag (otherwise you can not recompile query correctly)

Reply:
This will be done in a [separate ticket](https://st.yandex-team.ru/KIKIMR-20694) to speed up the development and split the big task into smaller ones for better workflow